product_packaging_unit_price_calculator/wizards/product_package_price.py

```diff
@@ -53,35 +69,42 @@ def _default_product_supplierinfo_id(self):
     @api.depends("packaging_price", "selected_packaging_id")
     def _compute_unit_price(self):
         if not self.selected_packaging_id:
-            self.unit_price = 0.0
+            self.unit_price = self.current_unit_price
         elif not self.selected_packaging_id.qty:
-            self.unit_price = 0.0
+            self.unit_price = self.current_unit_price
             self.warning_message = _(
                 "Unit price cannot be computed because the selected"
                 "packaging has no quantity set."
             )
         else:
             self.unit_price = self.packaging_price / self.selected_packaging_id.qty
             self.warning_message = " "
-            self._compute_package_prices()
+        self._compute_package_prices()
+
+    @api.depends(
+        "product_pricelist_item_id", "product_supplierinfo_id", "product_tmpl_id"
+    )
+    def _compute_current_unit_price(self):
+        """Compute the original unit price, the one  that the calculator will change."""
+        if self.product_pricelist_item_id:
```

Comment:
it would be better to have a loop on self here.

Reply:
I understand it is required on regular models, but here the compute is for the field of a wizard, we should always have one record, no ?